tests/test_integration.py

```diff
@@ -0,0 +1,196 @@
+import collections
+import threading
+import subprocess
+import os
+import json
+import mock
+import pwd
+import socket
+
+import pytest
+
+try:
+    from BaseHTTPServer import HTTPServer
+    from BaseHTTPServer import BaseHTTPRequestHandler
+except ImportError:
+    from http.server import HTTPServer
+    from http.server import BaseHTTPRequestHandler
+
+
+Request = collections.namedtuple('Request', ['command', 'path', 'headers', 'body'])
+
+
+class SentryHTTPServer(HTTPServer):
+    timeout = 0.1
+
+    def __init__(self, *args, **kwargs):
+        requests = kwargs.pop('requests')
+        HTTPServer.__init__(self, *args, **kwargs)
+        self.requests = requests
+
+    def handle_timeout(self):
+        pass
+
+
+class SentryHTTPRequestHandler(BaseHTTPRequestHandler):
+    def do_POST(self):
+        body_len = int(self.headers.get('Content-Length', '0'))
+        body = self.rfile.read(body_len)
+        request = Request(command=self.command, path=self.path, headers=dict(self.headers.items()), body=body)
+        self.server.requests.append(request)
+        self.send_response(200)
+        self.send_header('Content-Type', 'application/json')
+        body = json.dumps({'status': 'ok'}).encode('utf-8')
+        self.send_header('Content-Length', str(len(body)))
+        self.end_headers()
+        self.wfile.write(body)
+
+
+class TestHTTPServer(object):
```

Comment:
pytest gets mad about you having a class that starts with =Test= that isn't a test.